### PR TITLE
Fix terraform-validate hook

### LINKED ISSUE #31: terraform-validate breaks after 0.1.5
Execution of hook in 0.1.5

```
Terraform validate.......................................................Failed
- hook id: terraform-validate
- exit code: 1

Initializing modules...

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "null" (hashicorp/null) 2.1.2...
- Downloading plugin for provider "aws" (hashicorp/aws) 2.65.0...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.null: version = "~> 2.1"
* provider.template: version = "~> 2.1"

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

Error: Error in function call

  on terraform/modules/nsqd/data.tf line 2, in data "template_file" "opt_nsqd_etc_nsqd_config":
   2:   template = file(
   3: 
   4: 
    |----------------
    | path.module is "terraform/modules/nsqd"

Call to function "file" failed: no file exists at
terraform/templates/nsqd/nsqd_config.tpl.
```

Execution of hook in 0.1.8

```
Terraform validate.......................................................Failed
- hook id: terraform-validate
- exit code: 1

Terraform initialized in an empty directory!

The directory has no Terraform configuration files. You may begin working
with Terraform immediately by creating Terraform configuration files.

Error: Failed to read module directory

Module directory terraform does not exist or cannot be read.
```

Directory structure:

```
tree -L 2                                                                                                                                                    
.
├── README.md
└── terraform
    ├── bin
    ├── config
    ├── main.tf
    ├── Makefile
    ├── modules
    ├── templates
    ├── variables.tf
    └── versions.tf
```

### PR DESCRIPTION
**Pull request by yorinasub17**
Fixes #31

### CODE REVIEW

File: hooks/adoclink-check.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -e
+
+# OSX GUI apps do not pick up environment variables the same way as Terminal apps and there are no easy solutions,
+# especially as Apple changes the GUI app behavior every release (see https://stackoverflow.com/q/135688/483528). As a
+# workaround to allow GitHub Desktop to work, add this (hopefully harmless) setting here.
+export PATH=$PATH:/usr/local/bin
+
+if ! command -v asciidoc-link-check; then
```
Comment (brikis98) [inline hooks/adoclink-check.sh:10]: Is this related to this PR?

File: hooks/terraform-validate.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ export PATH=$PATH:/usr/local/bin
 
 for dir in $(echo "$@" | xargs -n1 dirname | sort -u | uniq); do
   pushd "$dir" >/dev/null
-  terraform init -backend=false "$dir"
-  terraform validate "$dir"
+  terraform init -backend=false
+  terraform validate
```
Comment (brikis98) [inline hooks/terraform-validate.sh:13]: Is `dir` not necessary because of the `pushd` above?

File: hooks/adoclink-check.sh (hunk 1 above)
Comment (yorinasub17) [reply hooks/adoclink-check.sh]: Doh! This is a WIP script that I did not intend to check in. Removed! 96f2c4f

File: hooks/terraform-validate.sh (hunk 2 above)
Comment (yorinasub17) [reply hooks/terraform-validate.sh:13]: Yup. In fact, `dir` is a relative path, which is why it is breaking.

**Comment by yorinasub17** [comment]
Thanks for review! I verified this works on one of our repos. Going to merge this in now!

**Pull request merged**